### PR TITLE
:children_crossing: Add new config api v2alpha1

### PR DESCRIPTION
**Pull request by blacksails**
The old API is messy and not well documented. This adds a new config API
while keeping backwards compatatibility with the old API.

### CODE REVIEW

File: api/config/v2alpha1/projectconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+	// GitCredentials holds a list of git credential configuration. The
+	// RepoPrefix of the GitCredential will be matched angainst repository URL in
+	// order to determine which credential to use. The GitCredential with the
+	// longest matching RepoPrefix will be selected.
+	GitCredentials []*GitCredential `json:"gitCredentials"`
+
+	// LogLevel sets the logging level of the controller.
+	//
+	// TODO: how does the level work after the introduction of the new logging
```
Comment (blacksails) [inline api/config/v2alpha1/projectconfig_types.go:57]: Write a proper description.

File: config/samples/config_v2alpha1_projectconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  labels:
+    app.kubernetes.io/name: projectconfig
+    app.kubernetes.io/instance: projectconfig-sample
+    app.kubernetes.io/part-of: styra-controller
+    app.kubernetes.io/managed-by: kustomize
+    app.kubernetes.io/created-by: styra-controller
+  name: projectconfig-sample
+spec:
+  # TODO(user): Add fields here
```
Comment (blacksails) [inline config/samples/config_v2alpha1_projectconfig.yaml:12]: Add full example

File: internal/config/config.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// Package config provides utilities for reading configfiles
```
Comment (blacksails) [inline internal/config/config.go:1]: Add license notice

File: api/config/v2alpha1/projectconfig_types_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			Ω(c.GetGitCredentialForRepo(repo)).To(Equal(expected))
+		},
+		Entry("empty list of git credentials", nil, "test", nil),
+
+		Entry("get eficode credentials",
+			[]*v2alpha1.GitCredential{
+				{
+					User:       "",
+					Password:   "",
+					RepoPrefix: "https://git.bankdata.eficode.io/",
```
Comment (simlau) [inline api/config/v2alpha1/projectconfig_types_test.go:41]: Change to some repo that is not bankdatas

**Review by simlau — CHANGES_REQUESTED** [review]

**Review by simlau — CHANGES_REQUESTED** [review]

**Review by simlau — APPROVED** [review]

**Pull request merged**